internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -245,6 +284,21 @@ func (r *ForeignClusterReconciler) checkJoined(fc *discoveryv1.ForeignCluster, f
 	return fc, nil
 }
 
+func (r *ForeignClusterReconciler) getOriginClusterSets() discoveryv1.OriginClusterSets {
+	if r.ForeignConfig != nil {
+		// testing
```

Comment:
As a general behavior, I would suggest to avoid adding testing branches when not strictly needed. In this case, for example, does the r.ForeignConfig != nil make sense only for testing?

Reply:
r.ForeignConfig != nil is true only in testing environment, because of slightly different behaviour. In this case I have not ClsuterConfig running, so I can't get policy on trusting CAs

Reply:
The problem is: let's suppose that there is a bug in the operator and the r.ForeignConfig is not Nil as expected when you are running this code outside testing, this would result in a error which we are treating as a testCase. I would suggest to refactor this and treat this unexpected parameter as an unexpected behavior.